src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAwaitUsingTests.cs

```diff
@@ -2395,5 +2377,71 @@ await using (new Disposable())
             var comp = CreateCompilationWithTasksExtensions(new[] { source, IAsyncDisposableDefinition }, options: TestOptions.ReleaseExe);
             CompileAndVerify(comp, expectedOutput: "StructAwaitable");
         }
+
+        [Fact, WorkItem(45111, "https://github.com/dotnet/roslyn/issues/45111")]
+        public void IAsyncDisposableInterfaceUnnecessaryForPatternDisposal()
+        {
+            var source = @"
+using System.Threading.Tasks;
+
+await using (new C()) { }
+
+class C
+{
+    public Task DisposeAsync()
+    {
+        System.Console.Write(""DISPOSED"");
+        return Task.CompletedTask;
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.MakeTypeMissing(WellKnownType.System_IAsyncDisposable);
```

Comment:
If we can test this case with `System.IAsyncDisposable` missing, rather than just hidden from `WellKnownTypes`, we should. It looks like `CreateCompilationWithTasksExtensions(source)` will not include `IAsyncDisposable`.

Reply:
Sure. I'm keeping `MakeTypeMissing` though, as the set of references included in our various TFMs grows over time.

Reply:
`MakeTypeMissing()` changes the compiler behavior though.

To verify the type is missing in the test, could we simply add `Assert.Null(comp.GetWellKnownType(WellKnownType.System_IAsyncDisposable);`?